File: backend/micro_epoweredmove_poi_chargingstation/src/main/java/com/kosdiam/epoweredmove/models/enums/Availability.java
package com.kosdiam.epoweredmove.models.enums;

public enum Availability {
    UNAVAILABLE("Unavailable"),
    AVAILABLE("Available"),
    INUSE("In use"),
    UNKNOWN("Unknown");

    private String availability;
    private Availability(String availability){
        this.availability = availability;
    }

    @Override
    public String toString(){
        return availability;
    }
}

File: frontend/E-PoweredMoveAndroidApp/app/src/main/java/com/kosdiam/epoweredmove/models/enums/Availability.kt
package com.kosdiam.epoweredmove.models.enums

import com.kosdiam.epoweredmove.R
import com.kosdiam.epoweredmove.models.interfaces.IEnumName

enum class Availability (override val stringResource: Int) : IEnumName {
    UNAVAILABLE(R.string.availability_unavailable),
    AVAILABLE(R.string.availability_available),
    INUSE(R.string.availability_inuse),
    UNKNOWN(R.string.availability_unknown);
}